utils_nlp/models/transformers/abstractive_summarization_bertsum.py

```
@@ -0,0 +1,832 @@
+                "Batch",
+                [
+                    "src",
+                    "segs",
+                    "mask_src",
+                    "tgt",
+                    "tgt_num_tokens",
+                    "src_str",
+                    "tgt_str",
+                ],
+            )
+            batch = Batch(
+                src=encoded_stories.to(device),
+                segs=encoder_token_type_ids.to(device),
+                mask_src=encoder_mask.to(device),
+                tgt_num_tokens=torch.stack(summary_num_tokens).to(device),
+                tgt=encoded_summaries.to(device),
+                src_str=stories,
+                tgt_str=summaries,
+            )
+        else:
+            Batch = namedtuple("Batch", ["src", "segs", "mask_src"])
+            batch = Batch(
+                src=encoded_stories.to(device),
+                segs=encoder_token_type_ids.to(device),
+                mask_src=encoder_mask.to(device),
+            )
+
+        return batch
+
+    def preprocess(self, story_lines, summary_lines=None):
+        """preprocess multiple data points
+
+           Args:
+              story_lines (list of strings): List of sentences.
+              targets (list of strings, optional): List of sentences.
+                  Defaults to None, which means it doesn't include summary and is
+                  not training data.
+
+            Returns:
+                If summary_lines is None, return list of list of token ids. Otherwise,
+                return a tuple of (list of list of token ids, list of list of token ids).
+
+        """
+        story_lines_token_ids = []
+        for line in story_lines:
+            try:
+                if len(line) <= 0:
+                    continue
+                story_lines_token_ids.append(
+                    self.tokenizer.encode(line, max_length=self.max_src_len)
+                )
+            except:
+                print(line)
+                raise
```

maybe raise ValueError("Invalid line: {}".format(line))

---

will add that